File: problems/13_1d_dp/0070_climbing_stairs/solution.swift
// 0070. Climbing Stairs
import Foundation
class Solution {
    func climbStairs(_ n: Int) -> Int {
        var a = 1, b = 1
        for _ in 0..<n { let t = a + b; a = b; b = t }
        return a
    }
}
let sol = Solution()
if sol.climbStairs(1) != 1 { fatalError("a") }
if sol.climbStairs(2) != 2 { fatalError("b") }
if sol.climbStairs(3) != 3 { fatalError("c") }
if sol.climbStairs(5) != 8 { fatalError("d") }
if sol.climbStairs(10) != 89 { fatalError("e") }
if sol.climbStairs(0) != 1 { fatalError("f") }
print("All tests passed!")

File: problems/13_1d_dp/0070_climbing_stairs/solution.dart
// 0070. Climbing Stairs

int climbStairs(int n) {
  var a = 1, b = 1;
  for (var i = 0; i < n; i++) {
    final t = a + b;
    a = b;
    b = t;
  }
  return a;
}

void main() {
  if (climbStairs(1) != 1) throw 'a';
  if (climbStairs(2) != 2) throw 'b';
  if (climbStairs(3) != 3) throw 'c';
  if (climbStairs(5) != 8) throw 'd';
  if (climbStairs(10) != 89) throw 'e';
  if (climbStairs(0) != 1) throw 'f';
  print('All tests passed!');
}
